### PR TITLE
server: implement mailer and send mail on invitation

### LINKED ISSUE #65: Implement a service for sending emails

### PR DESCRIPTION
**Pull request by BourgoisMickael**
Closes #65

### CODE REVIEW

File: server/src/config/config.js [hunk 1]
```diff
@@ -94,6 +94,51 @@ const config = convict({
...
+    fake: {
+      doc: 'True to use a fake smtp',
+      format: Boolean,
+      default: true,
+      env: 'MAILER_FAKE'
+    },
+    from: {
+      doc: 'Source email address',
+      format: String,
+      default: 'Raven-Build <build@raven-os.fr>',
```
Comment (Arignir) [inline server/src/config/config.js]:
```suggestion
      default: 'Raven-Build <build@raven-os.org>',
```

File: server/src/config/config.js [hunk 2]
```diff
@@ -94,6 +94,51 @@ const config = convict({
...
+      env: 'MAILER_FROM'
+    },
+    transporter: {
+      host: {
+        doc: 'Host of the smtp server',
+        format: String,
+        default: undefined,
```
Comment (Arignir) [inline server/src/config/config.js]: Default value are great to identify the format of the parameter

File: server/src/controller/invitation.js [hunk 3]
```diff
@@ -33,7 +33,14 @@ class InvitationController {
     })
       .save()
 
-    return invitation.toJSON()
+    invitation = invitation.toJSON()
+
+    const subject = 'Invitation build-raven-os-org'
+    const text = `Pour vous inscrire sur build-raven-os-org, utilisez le code suivant: ${invitation.uuid}`
```
Comment (Arignir) [inline server/src/controller/invitation.js]: In English please, and the name `build-raven-os-org` is quite technical, something like `Raven's Package Builder` or anything similar would probably be more meaningful.

**Review by Arignir — CHANGES_REQUESTED** [review]

File: server/src/config/config.js (hunk 2 above)
Comment (BourgoisMickael) [reply server/src/config/config.js]: the format is `String` and the default value is `undefined` there is no default value for smtp we use a fake smtp for develoment

Comment (Arignir) [reply server/src/config/config.js]: You can use `localhost`, and a format is different than a type, it describes what the string should contain.

Comment (BourgoisMickael) [reply server/src/config/config.js]: done

**Comment by Arignir** [comment]
(There's the port too :x)
done

**Review by Arignir — APPROVED** [review]

**Pull request merged**